nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,592 @@
+        self.true_action_indices = [self.action_space[f] for f in self.action_space_subset]
+
+    def action(self, action: Union[int, List[int]]):
+        if isinstance(action, int):
+            return self.true_action_indices[action]
+        else:
+            return [self.true_action_indices[a] for a in action]
+
+
+class CompilerGym(ExperimentFunction):
+    def __init__(self):
+        action_space_size = len(SmallActionSpaceLlvmEnv.action_space_subset)
+        parametrization = (
+            ng.p.Array(shape=(num_episode_steps,)).set_bounds(0, action_space_size - 1).set_integer_casting()
+        )
+        env = gym.make("llvm-ic-v0", observation_space="Autophase", reward_space="IrInstructionCountOz")
+        self.uris = list(env.datasets["benchmark://cbench-v1"].benchmark_uris())
+        self.compilergym_index = np.random.choice(self.uris)
+        env.reset(benchmark=self.compilergym_index)
+        super().__init__(self.eval_actions_as_list, parametrization=parametrization)
+
+    def make_env() -> gym.Env:
+        """Convenience function to create the environment that we'll use."""
+        # User the time-limited wrapper to fix the length of episodes.
+        env = gym.wrappers.TimeLimit(
+            env=SmallActionSpaceLlvmEnv(env=gym.make("llvm-v0", reward_space="IrInstructionCountOz")),
+            max_episode_steps=num_episode_steps,
+        )
+        env.require_dataset("cBench-v1")
+        env.unwrapped.benchmark = "cBench-v1/qsort"
+        return env
+
+    # @lru_cache(maxsize=1024)  # function is deterministic so we can cache results
+    def eval_actions(self, actions: Tuple[int]) -> float:
+        """Create an environment, run the sequence of actions in order, and return the
+        negative cumulative reward. Intermediate observations/rewards are discarded.
+
+        This is the function that we want to minimize.
+        """
+        with self.make_env() as env:
+            env.reset(benchmark=self.compilergym_index)
+            _, reward, _, _ = env.step(actions)
+        return -env.episode_reward
+
+    def eval_actions_as_list(self, actions: List[int]):
+        """Wrapper around eval_actions() that records the return value for later analysis."""
+        action_space_size = len(SmallActionSpaceLlvmEnv.action_space_subset)
+        reward = eval_actions(tuple(actions.tolist()))
+        # action_names = [SmallActionSpaceLlvmEnv.action_space_subset[a] for a in actions]
+        # print(len(rewards_list), f"{-reward:.6f}", " ".join(action_names), sep='\t')
+        return reward
+
+
+class GymMulti(ExperimentFunction):
+
+    env_names = GYM_ENV_NAMES
+
+    controllers = CONTROLLERS
+
+    ng_gym = [
+        "Copy-v0",
+        "RepeatCopy-v0",
+        "Reverse-v0",
+        "CartPole-v0",
+        "CartPole-v1",
+        "Acrobot-v1",
+        "FrozenLake-v0",
+        "FrozenLake8x8-v0",
+        "NChain-v0",
+        "Roulette-v0",
+    ]
+
+    def __init__(
+        self,
```

Comment:
all that follows seems to refer to the other gym envs so that should not be included here